src/main/java/Szczurki/Simulation/Setup/MapReader.java

```diff
@@ -8,44 +8,48 @@
 import java.io.BufferedReader;
 import java.io.FileReader;
 import java.io.IOException;
+import java.util.ArrayList;
 
 public class MapReader implements IMapReader {
 
+    ArrayList<String> labirynth = new ArrayList<>();
+
     public IEntity[][] getMap(SimulationSettings settings) {
 
         String fileName = settings.fileName;
-        IEntity[][] map = new IEntity[settings.mapWidth][settings.mapHeight];
+
         try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
 
-            String[] labirynth = new String[settings.mapHeight];
+            String line;
 
+            while((line = reader.readLine())!=null){
+                labirynth.add(line);
+            }
 
-            map = new IEntity[settings.mapWidth][settings.mapHeight];
 
-            for (int i = 0; i < settings.mapHeight; ++i) {
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
 
-                labirynth[i] = reader.readLine();
 
-            }
+        IEntity[][] map = new IEntity[labirynth.size()][labirynth.get(0).length()];
 
-            for (int i = 0; i < settings.mapHeight; ++i) {
-                for (int j = 0; j < settings.mapWidth; ++j) {
-                    if (labirynth[i].charAt(j) == '#') {
-                        map[j][i] = new Wall();
-                    }
-
-                    if (labirynth[i].charAt(j) == '@') {
-                        map[j][i] = new Obstacle(10, 10);
-                    }
+        for (int i = 0; i < labirynth.size(); ++i) {
+            for (int j = 0; j < labirynth.get(0).length(); ++j) {
+                if (labirynth.get(i).charAt(j) == '#') {
+                    map[j][i] = new Wall();
+                }
 
+                if (labirynth.get(i).charAt(j) == '@') {
+                    map[j][i] = new Obstacle(10, 10);
                 }
+
             }
-        } catch (IOException e) {
-            e.printStackTrace();
         }
+        int mapHeight=labirynth.size();
+        int mapWidth = labirynth.get(0).length();
```

Comment:
Te zmienne są nieużywane. Jaki był ich cel?  Jeśli po to żeby je wpisać do SimulationSettings to w tym momencie działa bez nich, więc bym się tym nie przejmował

Reply:
pozostałość po poprzednim koncepcie na to wczytywanie rozmiarów